SignalMessaging/Views/ThreadViewHelper.m

```diff
@@ -129,16 +129,18 @@ - (void)yapDatabaseModifiedExternally:(NSNotification *)notification
 
     DDLogVerbose(@"%@ %s", self.logTag, __PRETTY_FUNCTION__);
 
-    // External database modifications can't be converted into incremental updates,
-    // so rebuild everything.  This is expensive and usually isn't necessary, but
-    // there's no alternative.
-    [self.uiDatabaseConnection beginLongLivedReadTransaction];
-    [self.uiDatabaseConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
-        [self.threadMappings updateWithTransaction:transaction];
-    }];
-
-    [self updateThreads];
-    [self.delegate threadListDidChange];
+    if (self.shouldObserveDBModifications) {
```

Comment:
Small, safe optimization: we don't need to do this if we're not observing db modifications since we'll do it when we resume.